vespa/src/test/java/ai/marqo/search/SortByTest.java

```
@@ -0,0 +1,746 @@
+                                            hits, sortJsonWith4Fields, null, 10, 0))
+                    .isInstanceOf(RuntimeException.class)
+                    .hasMessageContaining(
+                            "sortBy fields cannot contain more than 3 sort fields. Found: 4");
+
+            // Build JSON with 5 fields to test the counter
+            String sortJsonWith5Fields =
+                    "[{\"field_name\":\"f1\",\"order\":\"asc\",\"missing\":\"last\"},"
+                            + "{\"field_name\":\"f2\",\"order\":\"desc\",\"missing\":\"first\"},"
+                            + "{\"field_name\":\"f3\",\"order\":\"asc\",\"missing\":\"last\"},"
+                            + "{\"field_name\":\"f4\",\"order\":\"desc\",\"missing\":\"first\"},"
+                            + "{\"field_name\":\"f5\",\"order\":\"asc\",\"missing\":\"last\"}]";
+
+            assertThatThrownBy(
+                            () ->
+                                    searcher.postProcessBySort(
+                                            hits, sortJsonWith5Fields, null, 10, 0))
+                    .isInstanceOf(RuntimeException.class)
+                    .hasMessageContaining(
+                            "sortBy fields cannot contain more than 3 sort fields. Found: 5");
+        }
+
+        @Test
+        void shouldHandleUnsupportedOrderValues() {
+            HybridSearcher searcher = new HybridSearcher();
+            HitGroup hits = createDummyHitGroup();
+
+            // Test unsupported order values - should default to ASC
+            String[] unsupportedOrders = {"ascending", "descending", "up", "down", "invalid", ""};
+
+            for (String order : unsupportedOrders) {
+                String sortJson =
+                        "[{\"field_name\":\"f1\",\"order\":\""
+                                + order
+                                + "\",\"missing\":\"last\"}]";
+
+                // Should not throw exception, should default to ASC
```

this is not a good practice. we should validate the input rigorously and fail fast

---

Revised this test